### PR TITLE
add default.url in settings, nested serializer and send image in test

### PR DESCRIPTION
**Pull request by Ostashko-Maryna**

### CODE REVIEW

File: Eshop/apps/galleries/models.py [hunk 1]
```diff
@@ -3,7 +3,10 @@
 from django.conf import settings
 from django.db import models
 
+from apps_generic.whodidit.models import WhoDidIt
 
+
+# class Gallery(WhoDidIt):
```
Comment (Ostashko-Maryna) [inline Eshop/apps/galleries/models.py:9]: ValueError at /galleries/
Cannot assign "<django.contrib.auth.models.AnonymousUser object at 0x04744C58>": "Gallery.created_by" must be a "User" instance.
Request Method: POST

...\feb_py_eshop\Eshop\apps_generic\whodidit\models.py in set_who_did_it
44                 instance.created_by = user 

...\feb_py_eshop\Eshop\apps\galleries\models.py in save
57   super().save(*args, **kwargs)

File: Eshop/apps/galleries/tests.py [hunk 2]
```diff
@@ -120,3 +129,114 @@ def test_get_gallery_failed(self):
...
+            'results': []
+        })
+
+
+    def test_patch_gallery(self):
+        p1 = Product.objects.create(name='other product', description='asd')
+        response = self.c.patch(
+            '/galleries/1/1/', 
+            data={
+                "product": p1.id,
```
Comment (oshevelo) [inline Eshop/apps/galleries/tests.py:193]:
```suggestion
                "product": {"id": p1.id},
```

File: Eshop/apps/galleries/serializers.py [hunk 3]
```diff
@@ -22,3 +20,10 @@ class GallerySerializer(serializers.ModelSerializer):
     class Meta:
         model = Gallery
         fields = ['product', 'id', 'name', 'image', 'url', 'size', 'size_x', 'size_y']
+
+
+    def to_representation(self, instance):
+        data = super().to_representation(instance)
+        data['product'] = ProductInGallerySerializer(
+            Product.objects.get(pk=data['product'])).data
```
Comment (oshevelo) [inline Eshop/apps/galleries/serializers.py:28]:
```suggestion
            Product.objects.get(pk=data['product']['id'])).data
```

File: Eshop/apps/galleries/tests.py [hunk 4]
```diff
@@ -12,37 +14,45 @@ class GalleryTestAPI(TestCase):
     def setUp(self):
         p1 = Product.objects.create(name='testp', description='asd')
         # p2 = Product.objects.create(name='testp', description='asd')
-        self.g1 = Gallery.objects.create(product=p1, name='test_g1', image='photo1.jpg')
-        self.g2 = Gallery.objects.create(product=p1, name='test_g2', image='photo2.jpg', size = 'max')
+
```
Comment (oshevelo) [inline Eshop/apps/galleries/tests.py:17]:
```suggestion
self.pass = 'password'
self.user = User.objects.create_user('user_nameasdadasdzzzzz', 'user_name@gmail.com',self.pass)
```

File: Eshop/apps/galleries/tests.py [hunk 5]
```diff
@@ -120,3 +129,114 @@ def test_get_gallery_failed(self):
...
+        self.assertEqual(response.json(), {
+            'count': 0, 
+            'next': None, 
+            'previous': None, 
+            'results': []
+        })
+
+
+    def test_patch_gallery(self):
+        p1 = Product.objects.create(name='other product', description='asd')
```
Comment (oshevelo) [inline Eshop/apps/galleries/tests.py:189]:
```suggestion
        p1 = Product.objects.create(name='other product', description='asd')
        self.c.login(username=self.user.username, password=self.pass)
```

**Pull request merged**